outputs.tf

```diff
@@ -1,5 +1,5 @@
 output "public_ip" {
-  value = "${aws_eip.default.public_ip}"
+  value = "${coalesce(join("", aws_eip.default.*.public_ip), aws_instance.default.public_ip)}"
```

Comment:
`coalesce(concat(`.... seems like the more appropriate pattern.

Reply:
Nevermind. @comeanother cleared up why you're using this pattern. `coalesce` does not operate on lists and `coalescelist` returns a list.